state/application.go

```diff
@@ -268,11 +268,18 @@ func (op *DestroyApplicationOperation) Build(attempt int) ([]txn.Op, error) {
 	case errAlreadyDying:
 		return nil, jujutxn.ErrNoOperations
 	case nil:
-		return ops, nil
-	}
-	if op.Force {
-		logger.Warningf("forcing destroy application for %v despite error %v", op.app.Name(), err)
-		return ops, nil
+		nErrors := len(op.Errors)
+		if nErrors == 0 {
+			return ops, nil
+		}
+		logger.Debugf("%d error%s encountered during %s removal", nErrors, plural(nErrors), op.app.Name())
+		if op.Force {
+			logger.Debugf("forcing application removal")
+			return ops, nil
+		}
+		// should be impossible to reach, as--by convention--we return an error
+		// and an empty ops slice when a force-able error occurs
+		err = errors.New("errors encountered, run again with --force")
```

Comment:
Errors from state should not include CLI instructions. We just return any error and it's up to the CLI to augment the error as needed.

Reply:
Updated